src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -169,6 +195,66 @@ JITValuePointer LLVMJITValue::div(JITValue& rh) {
       getValueTypeTag(), parent_function_, ans, "div", false);
 }
 
+JITValuePointer LLVMJITValue::div_with_error_check(JITValue& rh) {
+  LLVMJITValue& llvm_rh = static_cast<LLVMJITValue&>(rh);
+  checkOprandsType(this->getValueTypeTag(), rh.getValueTypeTag(), "div");
+
+  llvm::Type* rh_llvm_type =
+      getLLVMType(llvm_rh.getValueTypeTag(), parent_function_.getLLVMContext());
+
+  auto zero_const = rh_llvm_type->isIntegerTy()
+                        ? llvm::ConstantInt::get(rh_llvm_type, 0, true)
+                        : llvm::ConstantFP::get(rh_llvm_type, 0.);
+
+  llvm::BasicBlock* div_ok{nullptr};
+  llvm::BasicBlock* div_zero{nullptr};
+
+  div_ok = llvm::BasicBlock::Create(
+      parent_function_.getLLVMContext(), "div_ok", &parent_function_.func_);
+  div_zero = llvm::BasicBlock::Create(
+      parent_function_.getLLVMContext(), "div_zero", &parent_function_.func_);
+
+  getFunctionBuilder(parent_function_)
+      .CreateCondBr(
+          zero_const->getType()->isFloatingPointTy()
+              ? getFunctionBuilder(parent_function_)
+                    .CreateFCmp(llvm::FCmpInst::FCMP_ONE, llvm_rh.load(), zero_const)
+              : getFunctionBuilder(parent_function_)
+                    .CreateICmp(llvm::ICmpInst::ICMP_NE, llvm_rh.load(), zero_const),
+          div_ok,
+          div_zero);
+
+  getFunctionBuilder(parent_function_).SetInsertPoint(div_ok);
+
+  llvm::Value* ans = nullptr;
+  switch (getValueTypeTag()) {
+    case JITTypeTag::INT8:
+    case JITTypeTag::INT16:
+    case JITTypeTag::INT32:
+    case JITTypeTag::INT64:
+      ans = getFunctionBuilder(parent_function_).CreateSDiv(load(), llvm_rh.load());
+      break;
+    case JITTypeTag::FLOAT:
+    case JITTypeTag::DOUBLE:
+      ans = getFunctionBuilder(parent_function_).CreateFDiv(load(), llvm_rh.load());
+      break;
+    default:
+      LOG(FATAL) << "Invalid JITValue type for div operation. Name=" << getValueName()
+                 << ", Type=" << getJITTypeName(getValueTypeTag()) << ".";
+  }
+
+  getFunctionBuilder(parent_function_).SetInsertPoint(div_zero);
+  // CIDER_THROW(CiderException,"math op div zero error");
+  getFunctionBuilder(parent_function_)
+      .CreateRet(llvm::ConstantInt::get(
+          llvm::Type::getInt32Ty(parent_function_.getLLVMContext()),
+          Executor::ERR_DIV_BY_ZERO));
+  getFunctionBuilder(parent_function_).SetInsertPoint(div_ok);
```

Comment:
better to not include old cider files` Executor.h`. I think you can define  error code in 'Nextgen.h' or 'LLVMJITFunction.h'